src/bosh_azure_cpi/lib/cloud/azure/cloud.rb

```diff
@@ -333,7 +334,7 @@ def create_disk(size, cloud_properties, vm_cid = nil)
                 vmss_instance = @azure_client.get_vmss_instance(resource_group_name, instance_id_obj.vmss_name, instance_id_obj.vmss_instance_id)
                 location = vmss_instance[:location]
                 instance_type = vmss_instance[:sku][:name]
-                zone = vmss_instance[:zones][0]
+                zone = vmss_instance[:zones][0] unless vmss_instance[:zones].nil?
```

Comment:
When the VMSS' zones are nil?

BTW, do we support [both single-zone and zone-redundant scale sets](https://docs.microsoft.com/en-us/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-use-availability-zones)?

Reply:
if we did not specify the zones for the vmss, then the zones would be nil.

Reply:
and for the zone-redundant.
customer will have option to specify the vmss zones in the cloud properties.